parser-typechecker/src/Unison/Codebase/Causal.hs

```diff
@@ -35,181 +31,33 @@ where
 import Unison.Prelude
 
 import qualified Control.Monad.Extra as Monad (anyM)
-import Control.Monad.State (StateT)
-import qualified Control.Monad.State as State
 import qualified Control.Monad.Reader as Reader
+import qualified Control.Monad.State as State
 import qualified Data.Map as Map
-import Data.Sequence (ViewL (..))
-import qualified Data.Sequence as Seq
 import qualified Data.Set as Set
-import qualified U.Util.Cache as Cache
-import Unison.Hash (Hash)
-import Unison.Hashable (Hashable)
-import qualified Unison.Hashable as Hashable
+import Unison.Codebase.Causal.Type
+  ( Causal
+      ( UnsafeCons,
+        UnsafeMerge,
+        UnsafeOne,
+        currentHash,
+        head,
+        tail,
+        tails
+      ),
+    RawHash (RawHash, unRawHash),
+    pattern One,
+    pattern Cons,
+    pattern Merge,
+    before,
+    children,
+    head_,
+    lca,
+  )
+import qualified Unison.Hashing.V2.Convert as Hashing
+import Unison.Hashing.V2.Hashable (Hashable)
 import Prelude hiding (head, read, tail)
-import qualified Control.Lens as Lens
-
-{-
-`Causal a` has 5 operations, specified algebraically here:
-
-* `before : Causal m a -> Causal m a -> m Bool` defines a partial order on
-            `Causal`.
-* `head : Causal m a -> a`, which represents the "latest" `a` value in a causal
-          chain.
-* `one : a -> Causal m a`, satisfying `head (one hd) == hd`
-* `cons : a -> Causal a -> Causal a`, satisfying `head (cons hd tl) == hd` and
-          also `before tl (cons hd tl)`.
-* `merge : CommutativeSemigroup a => Causal a -> Causal a -> Causal a`, which is
-           commutative (but not associative) and satisfies:
-  * `before c1 (merge c1 c2)`
-  * `before c2 (merge c1 c2)`
-* `sequence : Causal a -> Causal a -> Causal a`, which is defined as
-              `sequence c1 c2 = cons (head c2) (merge c1 c2)`.
-  * `before c1 (sequence c1 c2)`
-  * `head (sequence c1 c2) == head c2`
--}
-
-newtype RawHash a = RawHash { unRawHash :: Hash }
-  deriving (Eq, Ord, Generic)
-
-instance Show (RawHash a) where
-  show = show . unRawHash
-
-instance Show e => Show (Causal m h e) where
-  show = \case
-    One h e      -> "One " ++ (take 3 . show) h ++ " " ++ show e
-    Cons h e t   -> "Cons " ++ (take 3 . show) h ++ " " ++ show e ++ " " ++ (take 3 . show) (fst t)
-    Merge h e ts -> "Merge " ++ (take 3 . show) h ++ " " ++ show e ++ " " ++ (show . fmap (take 3 . show) . toList) (Map.keysSet ts)
-
--- h is the type of the pure data structure that will be hashed and used as
--- an index; e.g. h = Branch00, e = Branch0 m
-data Causal m h e
```

Comment:
Is it necessary to split this module up right now? Seems like it's likely to cause us merge-conflict issues, would it be possible to do this refactor on trunk and merge it in instead?
Refactoring in a long-lived feature branch can cause a lot of headaches sometimes 😅